### PR TITLE
[Core] Fix unit management in QuantitySpinBox

### PR DESCRIPTION
**Pull request by 0penBrain**
https://forum.freecadweb.org/viewtopic.php?f=3&t=64261

### CODE REVIEW

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5232">pipeline status</a> for feature branch [PR_5232](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5232). Pipeline [425424885](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/425424885) was triggered at [29761cb](https://github.com/FreeCAD/FreeCAD/pull/5232/commits/29761cb6aa97f3c782547c9001a5a705d2cf12a9). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by wwmayer** [comment]
Does it work as expected for you to e.g. type "23 mm" into a quantity spin box and then use mouse wheel/spin buttons to increase to "24 mm". I tried it with your previous changes and there it instead switched to "1 mm".

**Comment by 0penBrain** [comment]
@wwmayer I have no such problem. I tested both with and without loosing focus between typing and using wheel mouse, and both behave OK.

File: src/Gui/QuantitySpinBox.cpp [hunk 1]
```diff
@@ -585,18 +585,19 @@ void QuantitySpinBox::openFormulaDialog()
...
+void QuantitySpinBox::updateFromCache(bool notify, bool updateUnit /* = true */)
 {
     Q_D(QuantitySpinBox);
     if (d->pendingEmit) {
         double factor;
         const Base::Quantity& res = d->cached;
-        QString text = getUserString(res, factor, d->unitStr);
+        auto tmpUnit(d->unitStr);
+        QString text = getUserString(res, factor, updateUnit?d->unitStr:tmpUnit);
         d->unitValue = res.getValue() / factor;
```
Comment (wwmayer) [inline src/Gui/QuantitySpinBox.cpp:601]: This might be a problem because `d->unitValue` should be in sync with `d->unitStr`.

**Comment by wwmayer** [comment]
OK, I will try the updated PR, then.

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5232">pipeline status</a> for feature branch [PR_5232](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5232). Pipeline [454035626](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/454035626) was triggered at [c7752fe](https://github.com/FreeCAD/FreeCAD/pull/5232/commits/c7752feddc4ee539765ce5bdebeb49b75b7dfdbf). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by wwmayer** [comment]
OK, everything works for me too now.

**Comment by wwmayer** [comment]
Merged as c4c81e6e69

**Pull request closed without merging**